packages/react-router/tests/router.test.ts

```diff
@@ -301,4 +301,45 @@ describe('encoding: splat param for /$', () => {
       'framework/react/guide/file-based-routing tanstack',
     )
   })
+
```

Comment:
Try adding this to [LN938 in router.ts](https://github.com/TanStack/router/blob/61ecd34f05c6dd2f13b2ad6490b1725cee2ff799/packages/react-router/src/router.ts#L938), it may just do the trick.

```tsx
      let pathname = dest.to
        ? this.resolvePathWithBase(fromPath, `${dest.to}`)
        : this.resolvePathWithBase(
            fromPath,
            stayingMatches?.find((d) => d.pathname === fromPath)?.routeId ||
              fromPath,
          )
```

Reply:
It works! Thanks!